Authenticator/authenticator.py

```diff
@@ -0,0 +1,20 @@
+from jupyterhub.auth import Authenticator
+from tornado import gen
+import csv
+
+
+class MyAuthenticator(Authenticator):
+
+    @gen.coroutine
+    def authenticate(self,handler,data):
+        csv_dict={}
+        with open('authentication.csv', mode='r') as csv_file:
+            csv_data = {row["title"]: row["value"] for row in csv.DictReader(csv_file, ("title", "value"))}
+            csv_dict.update(csv_data)
+
+        if csv_dict[username] == data['password']:
```

Comment:
What will happen when `username` is not in the csv dict? We might want to add a check for that. I think there are three cases:

1. username not found in the csv
2. username found, but password doesn't match
3. username found, password matches (success!)

It might be useful to add log statements to indicate which event occurred, e.g.

```python
# no such user
self.log.warning("No such user: %s", username)
# password mismatch
self.log.warning("Incorrect password for user: %s", username)
```

This will be useful to have in the logs, because we can see if the user tried to login, but failed.

Reply:
Done